.github/workflows/tests.yml

```diff
@@ -0,0 +1,34 @@
+name: Test
+on:
+  pull_request:
+  push: { branches: main }
+
+jobs:
+  test:
+    name: Run test suite
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout 
+        uses: actions/checkout@v2
+      
+      - name: Login to registry
+        run: echo "${{ github.token }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+
+      - name: Build and run image
+        run: |
+            IMG=ghcr.io/${{ github.repository }}
+            docker build -t ${IMG} .
+            docker push $IMG
+            docker run -v ${PWD}:/root ${IMG}
+
+      - name: Install pytest
+        run:  |
+            pip3 install -r requirements.txt
+            pip3 install pytest
+
+      - name: Run tests
+        run: |
+            pytest tests/bag_test.py 
+            pytest tests/processing_test.py
```

Comment:
I get the impression we're not actually using the container if we do it like this.
Yes, on line 23 we do the `docker run` command, but the following steps are I think still run on the github ubuntu VM?

Also, since we're already installing all the necessary requirements in the Dockerfile, we should not have to install them again here (if we were actually using the container to run the tests in).

Lastly, I don't think we want to run the `docker push` command in the test workflow. That seems like it should only happen in the workflow that runs when we tag a release.